frontend/src/features/properties/components/GeocoderAutoComplete.tsx

```diff
@@ -0,0 +1,128 @@
+import * as React from 'react';
+import { IGeocoderResponse, useApi } from 'hooks/useApi';
+import './GeocoderAutoComplete.scss';
+import { Form, FormControlProps } from 'react-bootstrap';
+import { DisplayError } from 'components/common/form';
+import ClickAwayListener from 'react-click-away-listener';
+import { debounce } from 'lodash';
+
+interface IGeocoderAutoCompleteProps {
+  field: string;
+  placeholder?: string;
+  disabled?: boolean;
+  autoSetting?: string;
+  required?: boolean;
+  debounceTimeout?: number;
+  value?: string;
+  onSelectionChanged?: (data: IGeocoderResponse) => void;
+  error?: any;
+  touch?: any;
+  onTextChange?: (value?: string) => void;
+}
+
+export const GeocoderAutoComplete: React.FC<IGeocoderAutoCompleteProps> = ({
+  field,
+  placeholder,
+  disabled,
+  autoSetting,
+  required,
+  onSelectionChanged,
+  debounceTimeout,
+  value,
+  touch,
+  error,
+  onTextChange,
+}) => {
+  const [options, setOptions] = React.useState<IGeocoderResponse[]>([]);
+  const api = useApi();
+
+  const search = debounce(async (val: string) => {
+    const data = await api.searchAddress(val);
```

Comment:
Ideally you should include any other relevant information on the form (i.e. City, Postal, Province) as part of the search.